lib/geo.js

```diff
@@ -11,75 +11,108 @@ var assert = require('assert');
+ * Filter a set of results using the given filters returned by `nearFilter()`.
+ * Can support multiple locations, but will include results from all of them.
+ *
+ * WARNING: "or" operator with GeoPoint does not work as expected, eg:
+ * {where: {or: [{location: {near: (29,-90)}},{name:'Sean'}]}}
+ * Will actually work as if you had used "and".  This is because geo filtering
+ * takes place outside of the SQL query, so the result set of "name = Sean" is
+ * returned by the database, and then the location filtering happens in the app
+ * logic.  So the "near" operator is always an "and" of the SQL filters, and "or"
+ * of other GeoPoint filters.
+ *
+ * Additionally, since this step occurs after the SQL result set is returned,
+ * if using GeoPoints with pagination the result set may be smaller than the
+ * page size.  The page size is enforced at the DB level, and then we may
+ * remove results at the Geo-app level.  If we "limit: 25", but 4 of those results
+ * do not have a matching geopoint field, the request will only return 21 results.
+ * This may make it erroneously look like a given page is the end of the result set.
  */
 
-exports.filter = function(arr, filter) {
-  var origin = filter.near;
-  var max = filter.maxDistance > 0 ? filter.maxDistance : false;
-  var key = filter.key;
-
-  // create distance index
+exports.filter = function(rawResults, filters) {
   var distances = {};
-  var result = [];
+  var results = [];
 
-  arr.forEach(function(obj) {
-    var loc = obj[key];
+  filters.forEach(function(filter) {
+    var origin = filter.near;
+    var max = filter.maxDistance > 0 ? filter.maxDistance : false;
+    var min = filter.minDistance > 0 ? filter.minDistance : false;
+    var unit = filter.unit;
+    var key = filter.key;
 
-    // filter out objects without locations
-    if (!loc) return;
+    // create distance index
+    rawResults.forEach(function(result) {
+      var loc = result[key];
 
-    if (!(loc instanceof GeoPoint)) {
-      loc = new GeoPoint(loc);
-    }
+      // filter out results without locations
+      if (!loc) return;
 
-    if (typeof loc.lat !== 'number') return;
-    if (typeof loc.lng !== 'number') return;
+      if (!(loc instanceof GeoPoint)) loc = GeoPoint(loc);
+      if (typeof loc.lat !== 'number') return;
+      if (typeof loc.lng !== 'number') return;
 
-    var d = GeoPoint.distanceBetween(origin, loc);
+      var d = GeoPoint.distanceBetween(origin, loc, {type: unit});
 
-    if (max && d > max) {
-      // dont add
-    } else {
-      distances[obj.id] = d;
-      result.push(obj);
-    }
-  });
+      // filter result if distance is either < minDistance or > maxDistance
+      if ((min && d < min) || (max && d > max)) return;
+      distances[result.id] = d;
+      results.push(result);
+    });
 
-  return result.sort(function(objA, objB) {
-    var a = objB[key];
-    var b = objB[key];
+    results.sort(function(resA, resB) {
```

Comment:
Could you please preserve argument names `objA` and `objB` to make the diff smaller and easier to review?